buildSrc/subprojects/binary-compatibility/src/main/groovy/org/gradle/binarycompatibility/transforms/FindGradleJar.groovy

```diff
@@ -17,25 +17,43 @@
 
 package org.gradle.binarycompatibility.transforms
 
-import org.gradle.api.artifacts.transform.ArtifactTransform
-import javax.inject.Inject
+import org.gradle.api.artifacts.transform.AssociatedTransformAction
+import org.gradle.api.artifacts.transform.InputArtifact
+import org.gradle.api.artifacts.transform.TransformAction
+import org.gradle.api.artifacts.transform.TransformOutputs
+import org.gradle.api.artifacts.transform.TransformParameters
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+
 import groovy.transform.CompileStatic
 
+@AssociatedTransformAction(FindGradleJarAction)
 @CompileStatic
-class FindGradleJar extends ArtifactTransform {
-    private final String target
+interface FindGradleJar {
+    @Input
+    String getTarget()
+    void setTarget(String target)
+}
```

Comment:
I still find it confusing from the user's point of view that sometimes we use the _verb-like_ name (e.g. `ExplodeZipAndFindJars`) to refer to the action itself, and sometimes (as in the case of `FindGradleJar`) we refer to the parameter object, while the actual action gets an `Action` suffix.